example_attributeencoder_test.go

```diff
@@ -0,0 +1,75 @@
+package netlink_test
+
+import (
+	"encoding/hex"
+	"fmt"
+	"log"
+
+	"github.com/mdlayher/netlink"
+)
+
+// nested is a nested structure within out.
+type nested struct {
+	A, B uint32
+}
+
+// out is an example structure we will use to pack netlink attributes.
+type out struct {
+	Number uint16
+	String string
+	Nested nested
+}
+
+// encodeNested is an example function used to adapt the ae.Do method
+// to encode an arbitrary structure.
+func (n nested) enocdeNested() func() ([]byte, error) {
```

Comment:
I'm not sold on the naming here, or if it should be a function on the type, or standalone.